File: C# OOP/Polymorphism-Lab/03.Shapes/Circle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Shapes
{
    public class Circle : Shape
    {
        public Circle(double radius)
        {
            Radius = radius;
        }

        public double Radius { get; private set; }

        public override double CalculateArea()
        {
            double result = (Math.PI * Radius * Radius);

            return result;
        }

        public override double CalculatePerimeter()
        {
            double result = Math.PI * 2 * Radius;

            return result;
        }
        public override string Draw()
        {
            return $"{base.Draw()} {GetType().Name}";
        }
    }
}

File: Java OOP/Polymorphism-Lab/src/shapes/Circle.java
package shapes;

public class Circle extends Shape implements CircleInterface{
    private Double radius;

    public Circle(Double radius) {
        setRadius(radius);
    }

    @Override
    Double calculatePerimeter() {
        return 2 * Math.PI * radius;
    }

    @Override
    Double calculateArea() {
        return Math.PI * radius * radius;
    }

    @Override
    public final Double getRadius() {
        return radius;
    }

    public void setRadius(Double radius) {
        this.radius = radius;
    }
}
